spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/Exchange.java

```diff
@@ -55,6 +55,17 @@
 	String autoDelete() default "false";
 
 	/**
+	 * @return true if the exchange is to be declared as internal.
+	 */
+	String internal() default "false";
```

Comment:
Ha! I almost added this option here when I was doing the original feature.
But than I decided to stay off it, just because an `internal` exchange does not make sense on the `@RabbitListener`, when we really would like to have some `binding` to the queue to retrieve messages.
Therefore we do allow to use an `internal` exchange by the client.

Am I missing anything in the `internal` exchange concept?